client/components/route/index.js

```diff
@@ -22,6 +22,11 @@ export function RouteProvider( {
 		() => ( { currentSection, currentRoute, currentQuery } ),
 		[ currentSection, currentRoute, currentQuery ]
 	);
+
+	if ( ! currentRoute ) {
```

Comment:
Question: What's the purpose of this change in the PR? Asking because this seems to be a significant change so I'd like to understand more about its impact 🙂

Reply:
We use `currentRoute` to determine something when rendering. However, it might be `undefined` (it's weird) on the first render and it leads to inconsistency.

For example, the following condition was false on the first render, so the component was loading the primary and recent sites on the first render. Later, the value became true, and the component changed to load all sites. Without this change, the dashboard might still show a couple of sites first and it looked weird.
* `shouldQueryAllSites: currentRoute && currentRoute !== '/jetpack/connect/authorize'`